lib/crypto/crypto_compat.c

```diff
@@ -62,9 +63,29 @@ crypto_compat_RSA_import(RSA ** key, BIGNUM * n, BIGNUM * e, BIGNUM * d,
 		(*key)->dmq1 = dmq1;
 		(*key)->iqmp = iqmp;
 	}
+#else
+	/* It's ok for d to be NULL here. */
```

Comment:
I'd be inclined to put `if (d == NULL)` first and have two `RSA_set0_key` calls, one of which takes an explicit NULL (with a comment indicating that NULL means we have only the public key).